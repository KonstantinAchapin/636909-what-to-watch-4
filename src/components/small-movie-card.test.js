import React from 'react';
import renderer from 'react-test-renderer';
import {SmallMovieCard} from './small-movie-card.jsx';

const onMouseEnter = () => {};

const films = [
  {title: `Fantastic Beasts`, src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`},
  {title: `Johnny English`, src: `img/johnny-english.jpg`},
  {title: `Macbeth`, src: `img/macbeth.jpg`},
  {title: `Midnight Special`, src: `img/midnight-special.jpg`},
  {title: `Mindhunter`, src: `img/mindhunter.jpg`},
  {title: `Pulp Fiction`, src: `img/pulp-fiction.jpg`},
  {title: `War of the worlds`, src: `img/war-of-the-worlds.jpg`},
  {title: `Snatch`, src: `img/snatch.jpg`}
];

describe(`small-movie-card`, () => {
  it(`SmallMovieCard string matching`, () => {
    const tree = renderer.create(films.map((film, index) => <SmallMovieCard key={film.title + index} filmTitle={film.title} filmImg={film.src} onMouseEnter={onMouseEnter}/>)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
