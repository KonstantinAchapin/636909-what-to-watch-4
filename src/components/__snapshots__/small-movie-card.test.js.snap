// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`small-movie-card SmallMovieCard string matching 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Johnny English"
        height="175"
        src="img/johnny-english.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Johnny English
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Macbeth"
        height="175"
        src="img/macbeth.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Macbeth
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Midnight Special"
        height="175"
        src="img/midnight-special.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Midnight Special
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Mindhunter"
        height="175"
        src="img/mindhunter.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Mindhunter
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Pulp Fiction"
        height="175"
        src="img/pulp-fiction.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Pulp Fiction
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="War of the worlds"
        height="175"
        src="img/war-of-the-worlds.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        War of the worlds
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Snatch"
        height="175"
        src="img/snatch.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Snatch
      </a>
    </h3>
  </article>,
]
`;
